sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -2817,4 +2817,42 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
       }
     }
   }
+
+  test("SPARK-29421: Create Table LIKE USING provider") {
+    val catalog = spark.sessionState.catalog
+    withTable("s", "t1", "t2", "t3", "t4", "t5") {
+      sql("CREATE TABLE s(a INT, b INT) USING parquet")
+      val source = catalog.getTableMetadata(TableIdentifier("s"))
+      assert(source.provider == Some("parquet"))
+
+      sql("CREATE TABLE t1 LIKE s USING orc")
+      val table1 = catalog.getTableMetadata(TableIdentifier("t1"))
+      assert(table1.provider == Some("orc"))
+
+      sql("CREATE TABLE t2 LIKE s USING com.databricks.spark.csv")
+      val table2 = catalog.getTableMetadata(TableIdentifier("t2"))
+      assert(table2.provider == Some("com.databricks.spark.csv"))
+
+      val e1 = intercept[ClassNotFoundException] {
+        sql("CREATE TABLE t3 LIKE s USING com.databricks.Spark.csv")
+      }.getMessage
+      assert(e1.contains("Failed to find data source: com.databricks.Spark.csv"))
+
+      val e2 = intercept[ClassNotFoundException] {
+        sql("CREATE TABLE t4 LIKE s USING unknown")
+      }.getMessage
+      assert(e2.contains("Failed to find data source"))
+
```

Comment:
How about tests for `NoSuchMethodException`? https://github.com/apache/spark/pull/26097/files#diff-a53c8b7022d13417a2ef33372464f9b5R80

Reply:
Oh, there is no `NoSuchMethodException `.
I will test `AnalysisException`, `ClassNotFoundException`, and `NoClassDefFoundError`

Reply:
`NoClassDefFoundError ` can happen here but it's hard to trigger in test here.